Make CI more dynamic (#3688)
demisto/demisto-sdk: CHANGELOG.md

## Unreleased
* Fixed an issue where unified integrations / scripts with a period in their name would not split properly.
* Improved logging for **lint** and **prepare-content** commands.

> * Internal: Updated the code to use `DOCKER_IO`, `CI_SERVER_HOST`, `CI_PROJECT_ID` environment variables.

Reviewer: ```suggestion
* Internal: Added the `DOCKER_IO`, `CI_SERVER_HOST`, `CI_PROJECT_ID` environment variables.
```

> * Internal: Added the `GCS_MARKET_BUCKET`, `GCS_MARKET_BUCKET_DEV`, `GCS_MARKET_XPANSE_BUCKET`, `GCS_MARKET_V2_BUCKET`, `GCS_MARKET_XSOAR_SAAS_BUCKET`, `DOCKER_IO`, `CI_PROJECT_ID`, `CI_PROJECT_ID` environment variables.

Author: ```suggestion
* Internal: Updated the code to use `DOCKER_IO`, `CI_SERVER_HOST`, `CI_PROJECT_ID` environment variables.
```

* Fixed an issue where **validate** falsely failed with error `DS108` on descriptions ending with new lines followed by square/curly brackets.
* Fixed an issue where **graph** commands would not clean their temporary files properly, causing successive commands to fail.
* Fixed an issue where an error log message changed the terminal color.

> * Enhanced configurability by introducing more dynamic settings.

Reviewer: ```suggestion
* Added the <env_var1>, <env_var2>... environment variables.
```
we should document all those new env vars,
but since this PR is urgent, you can do it in a separate PR.

> * Added the `GCS_MARKET_BUCKET`, `GCS_MARKET_BUCKET_DEV`, `GCS_MARKET_XPANSE_BUCKET`, `GCS_MARKET_V2_BUCKET`, `GCS_MARKET_XSOAR_SAAS_BUCKET`, `DOCKER_IO`, `CI_PROJECT_ID`, `CI_PROJECT_ID` environment variables.

Reviewer 2: If any of these are relevant for users, explain what they're for. Otherwise, add the `Internal:` prefix, and a short description for the usages.

> * Updated the code to use the following environment variables: GCS_MARKET_BUCKET, GCS_MARKET_BUCKET_DEV, GCS_MARKET_XPANSE_BUCKET, GCS_MARKET_V2_BUCKET, GCS_MARKET_XSOAR_SAAS_BUCKET, DOCKER_IO, CI_PROJECT_ID, CI_PROJECT_ID.

Reviewer: ```suggestion
*Added the `GCS_MARKET_BUCKET`, `GCS_MARKET_BUCKET_DEV`, `GCS_MARKET_XPANSE_BUCKET`, `GCS_MARKET_V2_BUCKET`, `GCS_MARKET_XSOAR_SAAS_BUCKET`, `DOCKER_IO`, `CI_PROJECT_ID`, `CI_PROJECT_ID`.
```
